src/compiler/tsc.ts

```diff
@@ -523,9 +532,11 @@ namespace ts {
             const parsedCommandLine = parseConfigFile();
             const newFileNames = ts.map(parsedCommandLine.fileNames, compilerHost.getCanonicalFileName);
             const canonicalRootFileNames = ts.map(rootFileNames, compilerHost.getCanonicalFileName);
+            const newTypeNames = getTypeNames(parsedCommandLine.options);

```

Comment:
this means that if the types are outside the project directory, we will not catch their change events.